docs/advanced_topics/customisation/page_editing_interface.rst

```diff
@@ -57,22 +57,99 @@ Wagtail provides a general-purpose WYSIWYG editor for creating rich text content
 
 However, template output from :class:`~wagtail.wagtailcore.fields.RichTextField` is special and need to be filtered to preserve embedded content. See :ref:`rich-text-filter`.
 
-If you're interested in extending the capabilities of the Wagtail WYSIWYG editor (``hallo.js``), See :ref:`extending_wysiwyg`.
+
+.. _rich_text_features:
+
+Limiting features in a rich text field
+--------------------------------------
+
+By default, the rich text editor provides users with a wide variety of options for text formatting and inserting embedded content such as images. However, we may wish to restrict a rich text field to a more limited set of features - for example:
+
+ * The field might be intended for a short text snippet, such as a summary to be pulled out on index pages, where embedded images or videos would be inappropriate;
+ * When page content is defined using :ref:`StreamField <streamfield>`, elements such as headings, images and videos are usually given their own block types, alongside a rich text block type used for ordinary paragraph text; in this case, allowing headings and images to also exist within the rich text content is redundant (and liable to result in inconsistent designs).
+
+This can be achieved by passing a ``features`` keyword argument to ``RichTextField``, with a list of identifiers for the features you wish to allow:
+
+.. code-block:: python
+
+    body = RichTextField(features=['p', 'h2', 'h3', 'b', 'i', 'link'])
+
+The recognised feature identifiers are as follows (note that add-on modules may add to this list):
```

Comment:
Passing identifiers only, it's not possible to configure features further than "enabled/disabled", contrary to what `WAGTAILADMIN_RICH_TEXT_EDITORS` allows. Not sure how big of a deal this is. An example use case would be wanting to limit the level of nesting of `ul` list items to 0 only for some fields.

I think the obvious trade-off is that this would make the `features` definitions less editor-agnostic, because not all editors will provide the same degree of configuration. But passing per-field options to features still seems desirable.